protocol/avdl/keybase1/gui_file_access.avdl

```diff
@@ -0,0 +1,160 @@
+@namespace("keybase.1")
+
+protocol GuiFileAccess {
+
+  import idl "common.avdl";
+
+/*
+  File manipulation interface
+  
+  Caller provides OpID to asynchronous method call, then polls 
+  by calling a get* method until status is no
+  longer pending or operation is cancelled.
+
+  Caller can optionally block by calling wait
+ 
+  KBFS paths are expected to start with "private" or "public" - mount name is implied.
+  Other paths are absolute, full paths.
+*/
+
+  // Random GUID provided by caller 
+  fixed OpID(16);
+  
+  enum ItemType {
+    FILE,
+    DIR,
+    SYM,
+    EXEC
+  }
+
+  record Item {
+    string path;
+    Time time;
+    int size;   // size of file or count of dir entries (TBD - mmay be expensive/unnecessary)
+    ItemType itemType;
+  }
+
+  @typedef("int")
+  record ErrornNum {}
+
+  enum Status {
+    finished,
+    pending,
+    invalidated,
+    error
+  }
+
+  record Result {
+    Status status;
+    int progress;       // if status == pending
+    ErrornNum errnum;   // if status == error
+    string description; // if status == error
+  }
+
+  record ListResult {
+    array<Item> items;
+    Result result;
+  }
+
+  record FileContent {
+    bytes data
+    Result result;
+  }
+
+  /**
+   Begin list if items in directory at path
+   Retrieve results with getList()
+   Can be a single file to get flags/status
+   */
+  ErrorNum list(OpID opID, string path);
+
+  /**
+   Begin recursive list if items in directory at path
+   */
+  ErrorNum listRecursive(OpID opID, string path);
+
+  /**
+   Get list if items in progress. Can indicate status of pending
+   to get more items.
+   */
+  ListResult getList(OpID opID);
+
+  /**
+   Begin copy of file or directory 
+   */
+  ErrorNum copy(OpID opID, string src, string dest);
+
+  /**
+   Begin recursive copy of directory 
+   */
+  ErrorNum copyRecursive(OpID opID, string src, string dest);
+
+  /**
+   Begin move of file or directory 
+   */
+  ErrorNum move(OpID opID, string src, string dest);
+
+  /**
+   Begin recursive move of directory 
+   */
+  ErrorNum moveRecursive(OpID opID, string src, string dest);
+
+  /**
+   Rename file or directory
+   */
+  ErrorNum rename(string src, string dest);
```

Comment:
On unix `mv` and `rename` are the same command, but behave differently if it is within a partition or across a partition.  I wonder if we want to break out both here.  It might be confusing to people to know which one to use...